hack/tests/e2e-helm.sh

```diff
@@ -1,157 +1,29 @@
-    kubectl delete --ignore-not-found=true namespace ${test_namespace}
-    kubectl delete --ignore-not-found=true clusterrolebinding nginx-operator-system-metrics-reader
-    make undeploy
-}
-
-operator_logs() {
-    kubectl get all --namespace=${operator_namespace}
-    kubectl get events --namespace=${operator_namespace}
-    kubectl logs deployment/nginx-operator-controller-manager -c manager --namespace=${operator_namespace}
-}
-
-test_operator() {
-    # kind has an issue with certain image registries (ex. redhat's), so use a
-    # different test pod image.
-    local metrics_test_image="$METRICS_TEST_IMAGE"
-
-    # wait for operator pod to run
-    if ! timeout 1m kubectl rollout status deployment/nginx-operator-controller-manager -n ${operator_namespace} ;
-    then
-        error_text "Failed to rollout status"
-        operator_logs
-        exit 1
-    fi
-
-    metrics_service="nginx-operator-controller-manager-metrics-service"
-
-    # verify that metrics service was created
-    if ! timeout 60s bash -c -- "until kubectl get service/${metrics_service} --namespace=${operator_namespace} > /dev/null 2>&1; do sleep 1; done";
-    then
-        error_text "Failed to get metrics service"
-        operator_logs
-        exit 1
-    fi
-
-
-    serviceaccount_secret=$(kubectl get serviceaccounts default -n ${operator_namespace} -o jsonpath='{.secrets[0].name}')
-    token=$(kubectl get secret ${serviceaccount_secret} -n ${operator_namespace} -o jsonpath={.data.token} | base64 -d)
-
-    # verify that the metrics endpoint exists
-    if ! timeout 60s bash -c -- "until kubectl run --attach --rm --restart=Never --namespace=${operator_namespace} test-metrics --image=${metrics_test_image} -- -sfkH \"Authorization: Bearer ${token}\" https://${metrics_service}:8443/metrics; do sleep 1; done";
-    then
-        error_text "Failed to verify that metrics endpoint exists"
-        operator_logs
-        exit 1
-    fi
+# create test directories
+test_dir=./test
+tests=$test_dir/e2e-helm
 
-    # create CR
-    kubectl create --namespace=${test_namespace} -f config/samples/helm.example_v1alpha1_nginx.yaml
-    trap_add "kubectl delete --namespace=${test_namespace} --ignore-not-found -f ${OPERATORDIR}/config/samples/helm.example_v1alpha1_nginx.yaml" EXIT
-    if ! timeout 1m bash -c -- "until kubectl get --namespace=${test_namespace} nginxes.helm.example.com nginx-sample -o jsonpath='{..status.deployedRelease.name}' | grep 'nginx-sample'; do sleep 1; done";
-    then
-        error_text "Failed to create CR"
-        operator_logs
-        exit 1
-    fi
+export TRACE=1
+export GO111MODULE=on
 
+# set default envvars
+setup_envs $tmp_sdk_root
 
+# download and load required images for the tests
+docker pull $KUBE_RBAC_PROXY_IMAGE
+load_image_if_kind $KUBE_RBAC_PROXY_IMAGE
+docker pull $CURL_IMAGE
+load_image_if_kind $CURL_IMAGE
```

Comment:
Is it necessary to pull and load these images into `kind`? Since they're available in remote repos, I would think `kind` could pull them no problem.

Reply:
Yes, we need to pull and load the images otherwise it does not work.

Reply:
See: https://github.com/kubernetes-sigs/kind/issues/1202#issuecomment-568557667